### ISSUE TITLE
Internal Error Win64

### ISSUE THREAD

**Issue by roigcarlo** [KratosMultiphysics/Kratos#895]
@pooyan-dadvand  Windows compilation is throwing an internal error in this file:
tetrahedra_mesh_worst_element_smoothing_process.cpp

That's all I know

**Comment by pooyan-dadvand** [comment]
Would you please tell me more or less in which method or line?

**Comment by roigcarlo** [comment]
Looking into it

**Comment by roigcarlo** [comment]
Lines 79:
```c++
MathUtils<double>::CrossProduct(normal, v1, v2);
```

**Comment by roigcarlo** [comment]
Every line with the variable `normal` is causing the error.

Pretty sure is because in https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/math_utils.h#L829

`CrossProduct` Is defined as:
```C++
static inline void CrossProduct(
        array_1d<double, 3>& c, 
        const array_1d<double, 3>& a, 
        const array_1d<double, 3>& b
        )
    {
        c[0] = a[1]*b[2] - a[2]*b[1];
        c[1] = a[2]*b[0] - a[0]*b[2];
        c[2] = a[0]*b[1] - a[1]*b[0];
    }
```

And point is no longer a template of `<double, 3>` while calling it in: 

```C++
...
Point normal;
MathUtils<double>::CrossProduct(normal, v1, v2);
```

Probably the same happens in the other lines as it uses the overloaded operators for add divide etc...

**Comment by pooyan-dadvand** [comment]
`Point` is an `array_1d` but you can directly use the `array_1d` instead of point to see if it compiles...

**Comment by roigcarlo** [comment]
Nop, same problems... But still problems are only in the lines with the `normal` variable

**Comment by roigcarlo** [comment]
Changing the name also does not solve the problem, so can't be shadowing...

**Comment by pooyan-dadvand** [comment]
Making the loop instead of template assignment works?

**Comment by roigcarlo** [comment]
Nop. But anyway this `MathUtils<double>::CrossProduct(normal, v1, v2);` still has no template loops and also breaks

**Comment by roigcarlo** [comment]
Some updates, I've been doing tests and I found that:

While trying to change the code to check if compiles i saw that all errors with normal happens after the face is defined. Pulling the string;

```C++
// Triangle3D3<Point > face(rTetrahedra(tetrahedra_connectivity[i][0]), rTetrahedra(tetrahedra_connectivity[i][1]), rTetrahedra(tetrahedra_connectivity[i][2]));
```

Commenting this line and making some changes in the code makes the normal lines to compile again. 
honestly I have no clue about what is happening :S

**Comment by pooyan-dadvand** [comment]
Would you please copy the entire method here?

**Comment by roigcarlo** [comment]
```C++
void TetrahedraMeshWorstElementSmoothingProcess::CalculateElementOptimumPosition(NodeType& rNode, Geometry<Node<3> > const& rTetrahedra, Point& rOptimumPoint) {
		std::size_t i = 0;

		for (; i < 4; i++)
			if (rNode.Id() == rTetrahedra[i].Id())
				break;
		constexpr int tetrahedra_connectivity[4][3] = { {3,2,1},{2,3,0},{0,3,1},{0,1,2} };
		Triangle3D3<Point > face(rTetrahedra(tetrahedra_connectivity[i][0]), rTetrahedra(tetrahedra_connectivity[i][1]), rTetrahedra(tetrahedra_connectivity[i][2]));
		Point center = face.Center();
		Point v1 = face[0] - face[1];
		Point v2 = face[0] - face[2];
		Point normal;
		MathUtils<double>::CrossProduct(normal, v1, v2);
		double norm = norm_2(normal);
		if(norm > std::numeric_limits<double>::epsilon())
		normal /= norm;
		constexpr double height_coeficient = 0.81649658092772603273242802490196; // sqrt(6.00)/3.00
		rOptimumPoint = center + normal * face.AverageEdgeLength() * height_coeficient;
	}
```

**Comment by pooyan-dadvand** [comment]
But you are using the face. How it compiles commenting it?

**Comment by roigcarlo** [comment]
Because I replaced the face with some pre-built values. Then the problem dissapears (of coruse the code is not correct)

**Comment by pooyan-dadvand** [comment]
Try this:

```C++
void TetrahedraMeshWorstElementSmoothingProcess::CalculateElementOptimumPosition(NodeType& rNode, Geometry<Node<3> > const& rTetrahedra, Point& rOptimumPoint) {
		std::size_t i = 0;

		for (; i < 4; i++)
			if (rNode.Id() == rTetrahedra[i].Id())
				break;
		constexpr int tetrahedra_connectivity[4][3] = { {3,2,1},{2,3,0},{0,3,1},{0,1,2} };
		Triangle3D3<Point > face(rTetrahedra(tetrahedra_connectivity[i][0]), rTetrahedra(tetrahedra_connectivity[i][1]), rTetrahedra(tetrahedra_connectivity[i][2]));
		Point center(0.5,0.5,0.5);
		Point normal = face.Normal(center)
		constexpr double height_coeficient = 0.81649658092772603273242802490196; // sqrt(6.00)/3.00
		rOptimumPoint = center + normal * face.AverageEdgeLength() * height_coeficient;
	}
```

**Comment by roigcarlo** [comment]
Nope, same problem

**Comment by roigcarlo** [comment]
@RiccardoRossi Found the problem. It was caused by the ```constexpr ```.

**Comment by RiccardoRossi** [comment]
can we close?

**Comment by roigcarlo** [comment]
It will be closed automatically when we merge #909
